.github/workflows/main.yml

```diff
@@ -69,34 +80,28 @@ jobs:
           # Simply keep the database file but remove the repo.
           cp stave/simple-backend/db.sqlite3 .
           rm -rf stave
-      - name: Install deep learning frameworks
+      - name: Test backbone Forte import test
         run: |
           pip install --progress-bar off torch==${{ matrix.torch-version }}
-          pip install --progress-bar off tensorflow==${{ matrix.tensorflow-version }}
-      - name: Install Texar
-        run: |
-          git clone https://github.com/asyml/texar-pytorch.git
-          cd texar-pytorch
+          # Try to install Forte backbone only and test basic imports.
           pip install --use-feature=in-tree-build --progress-bar off .
-          cd ..
-          # Remove them to avoid confusion.
-          rm -rf texar-pytorch
+          pytest tests/forte/texar_nondependency_test.py
+          pip uninstall -y torch
+      - name: Test modules need extra packages
+        run: |
+          pip install --use-feature=in-tree-build --progress-bar off .[test]
+          if [ ${{ matrix.modules.module }} != "" ]
+          then
+            pip install --use-feature=in-tree-build --progress-bar off .[${{ matrix.modules.module }}]
+            pip install --progress-bar off pytest==5.1.3 coverage
```

Comment:
I think we have mentioned this, we shouldn't do the matrix test together with the main test, this will make the CI time really long because the normal test is also done N times (N is the number of matrix combinations).